### PR TITLE
fix: min max validation without native input[type=time] support

### PR DESCRIPTION
**Pull request by pixelbandito**
This fixes issue #254

### CODE REVIEW

File: src/components/InputTime/utils.ts [hunk 1]
```diff
@@ -152,3 +152,34 @@ export const guessTimeFromString = (string: string) => {
   const time = getShortTimeString(hours, minutes);
   return hasValidUserInput ? { time, hours, minutes } : {};
 };
+
+const getTotalMinutes = (timeString?: string) => {
+  const [hours, minutes] = (timeString || ':').split(':');
+  const totalMinutes = parseInt(hours, 10) * 60 + parseInt(minutes, 10);
+  return !Number.isNaN(totalMinutes) ? totalMinutes : undefined;
+};
```
Comment (wendigolabs) [inline src/components/InputTime/utils.ts:160]: Cleverly terse.

**Review by wendigolabs — APPROVED** [review]
🌮

**Review by bkonuwa — APPROVED** [review]

File: src/components/InputTime/utils.ts (hunk 1 above)
Comment (pixelbandito) [reply src/components/InputTime/utils.ts:160]: I experienced a surge of unexpected pride.
I don't think anyone's ever said that about me before, in any context. :)

File: src/components/InputTime/utils.ts [hunk 2]
```diff
@@ -152,3 +152,34 @@ export const guessTimeFromString = (string: string) => {
...
+  let validationMessage = '';
+  const totalMinutes = getTotalMinutes(value || ':');
+  const maxTotalMinutes = getTotalMinutes(max);
+  const minTotalMinutes = getTotalMinutes(min);
+
+  if (totalMinutes && maxTotalMinutes && totalMinutes > maxTotalMinutes) {
+    validationMessage = `Value must be ${max} or earlier.`;
+  }
+
+  if (totalMinutes && minTotalMinutes && totalMinutes < minTotalMinutes) {
```
[the remark this answers is not in the document]
Comment (pixelbandito) [reply src/components/InputTime/utils.ts:180]: I see what you're saying. It looks like this code will be wrong if total minutes is 0, even if there's a min validation set.

I'll verify that and get a fix out right away. Thanks for the catch!

**Pull request merged**